tests/datastructures/test_background_task.py

```diff
@@ -23,3 +23,24 @@ def handler() -> None:
         response = client.get("/")
         assert response.status_code == HTTP_200_OK
         assert values == [1, 2, 3, 4, 5, 6]
+
+
+async def test_background_tasks_task_group_execution() -> None:
+    values: List[int] = []
+
+    def extend_values(values_to_extend: List[int]) -> None:
+        values.extend(values_to_extend)
+
+    tasks = BackgroundTasks(
+        [BackgroundTask(extend_values, [1, 2, 3]), BackgroundTask(extend_values, values_to_extend=[4, 5, 6])],
+        run_in_task_group=True,
+    )
+
+    @get("/", background=tasks)
+    def handler() -> None:
+        return None
+
+    with create_test_client(handler) as client:
+        response = client.get("/")
+        assert response.status_code == HTTP_200_OK
+        assert values == [1, 2, 3, 4, 5, 6]
```

Comment:
Either sort the list or use a set as the order of the values in the list is just incidental.